File: packages/shared-kmp/core/analytics/src/commonMain/kotlin/dev/forcetower/melon/core/analytics/AnalyticsKeys.kt
package dev.forcetower.melon.core.analytics

// With only a couple of generic events, the analysis lives in these property
// *values* — so a typo here would silently split a funnel. Keep the known
// values as constants rather than inline strings at call sites. These are also
// the cross-platform contract: iOS mirrors the same strings.

object Screens {
    // Onboarding (outer nav)
    const val WELCOME = "welcome"
    const val INTRO = "intro"
    const val LOGIN = "login"
    const val SYNC = "sync"
    const val READY = "ready"
    const val FOLIO_RUNNER = "folio_runner"

    // Tab roots
    const val OVERVIEW = "overview"
    const val DISCIPLINES = "disciplines"
    const val SCHEDULE = "schedule"
    const val MESSAGES = "messages"
    const val ME = "me"

    // Pushed screens
    const val DISCIPLINE_DETAIL = "discipline_detail"
    const val MESSAGE_DETAIL = "message_detail"
    const val SETTINGS = "settings"
    const val PASSKEYS = "passkeys"
    const val CALENDAR = "calendar"
    const val FINAL_COUNTDOWN = "final_countdown"
    const val LICENSES = "licenses"

    const val MATERIALS = "materials"
    const val MATERIALS_DISCIPLINE = "materials_discipline"
    const val MATERIALS_DETAIL = "materials_detail"
    const val MATERIALS_SAVED = "materials_saved"

    const val PARADOXO = "paradoxo"
    const val PARADOXO_DISCIPLINE = "paradoxo_discipline"
    const val PARADOXO_TEACHER = "paradoxo_teacher"
    const val PARADOXO_EXPLORE = "paradoxo_explore"

    const val ENROLLMENT = "enrollment"
    const val ENROLLMENT_OFFERS = "enrollment_offers"
    const val ENROLLMENT_DISCIPLINE = "enrollment_discipline"
    const val ENROLLMENT_TIMETABLE = "enrollment_timetable"
    const val ENROLLMENT_REVIEW = "enrollment_review"
    const val ENROLLMENT_SUCCESS = "enrollment_success"

    const val CAMPUS_EVENT = "campus_event"
    const val CAMPUS_EVENT_ACTIVITY = "campus_event_activity"
    const val CAMPUS_EVENT_SPEAKERS = "campus_event_speakers"
    const val CAMPUS_EVENT_WORKSHOPS = "campus_event_workshops"
    const val CAMPUS_EVENT_VENUES = "campus_event_venues"
    const val CAMPUS_EVENT_ORGANIZATIONS = "campus_event_organizations"
}

object ContentTypes {
    const val HUB = "hub"
    const val CTA = "cta"
    const val TILE = "tile"
    const val SHORTCUT = "shortcut"

    const val DISCIPLINE = "discipline"
    const val SEMESTER = "semester"
    const val SCHEDULE_DAY = "schedule_day"
    const val CALENDAR_EVENT = "calendar_event"
    const val MESSAGE = "message"
    const val ACTIVITY = "activity"

    const val MATERIAL = "material"
    const val PARADOXO_ENTITY = "paradoxo_entity"
    const val PARADOXO_EXPLORE = "paradoxo_explore"
    const val OFFER = "offer"
    const val ENROLLMENT = "enrollment"

    const val DOCUMENT = "document"
    const val SETTING = "setting"
    const val PASSKEY = "passkey"
}

File: apps/ios/UNESKit/Sources/UNESKit/Analytics/AnalyticsKeys.swift
// With only a couple of generic events, the analysis lives in these property
// *values* — a typo here would silently split a funnel. Keep the known values
// as constants rather than inline strings at call sites. These mirror the
// Android contract in `packages/shared-kmp/.../analytics/AnalyticsKeys.kt`
// verbatim so the two platforms never drift; add new values there first.

enum Screens {
    // Onboarding (outer nav)
    static let welcome = "welcome"
    static let intro = "intro"
    static let login = "login"
    static let sync = "sync"
    static let ready = "ready"
    static let folioRunner = "folio_runner"

    // Tab roots
    static let overview = "overview"
    static let disciplines = "disciplines"
    static let schedule = "schedule"
    static let messages = "messages"
    static let me = "me"

    // Pushed screens
    static let disciplineDetail = "discipline_detail"
    static let messageDetail = "message_detail"
    static let settings = "settings"
    static let passkeys = "passkeys"
    static let calendar = "calendar"
    static let finalCountdown = "final_countdown"
    static let licenses = "licenses"

    static let materials = "materials"
    static let materialsDiscipline = "materials_discipline"
    static let materialsDetail = "materials_detail"
    static let materialsSaved = "materials_saved"

    static let retrospective = "retrospective"
    static let paradoxo = "paradoxo"
    static let paradoxoDiscipline = "paradoxo_discipline"
    static let paradoxoTeacher = "paradoxo_teacher"
    static let paradoxoExplore = "paradoxo_explore"

    static let enrollment = "enrollment"
    static let enrollmentOffers = "enrollment_offers"
    static let enrollmentDiscipline = "enrollment_discipline"
    static let enrollmentTimetable = "enrollment_timetable"
    static let enrollmentReview = "enrollment_review"
    static let enrollmentSuccess = "enrollment_success"

    static let campusEvent = "campus_event"
    static let campusEventActivity = "campus_event_activity"
    static let campusEventSpeakers = "campus_event_speakers"
    static let campusEventWorkshops = "campus_event_workshops"
    static let campusEventVenues = "campus_event_venues"
    static let campusEventOrganizations = "campus_event_organizations"
}

enum ContentTypes {
    static let hub = "hub"
    static let cta = "cta"
    static let tile = "tile"
    static let shortcut = "shortcut"

    static let discipline = "discipline"
    static let semester = "semester"
    static let scheduleDay = "schedule_day"
    static let calendarEvent = "calendar_event"
    static let message = "message"
    static let activity = "activity"

    static let material = "material"
    static let paradoxoEntity = "paradoxo_entity"
    static let paradoxoExplore = "paradoxo_explore"
    static let offer = "offer"
    static let enrollment = "enrollment"

    static let document = "document"
    static let setting = "setting"
    static let passkey = "passkey"
}
